packages/core/parcel/src/cli.js

```diff
@@ -343,15 +343,27 @@ async function normalizeOptions(command): Promise<InitialParcelOptions> {
   }
 
   let serve = false;
-  let {port, host} = command;
+  let {port = 1234, host} = command;
+
+  // Throw an error if port value is invalid...
+  if (!Number.isInteger(port)) {
+    throw new Error(`Defined port is not a valid integer`);
```

Comment:
```suggestion
    throw new Error(`Defined port '${port}' is not a valid integer`);
```

Reply:
yeah tried that but we parse it so it would always return `NaN`

Reply:
We could do this:
```diff
diff --git packages/core/parcel/src/cli.js packages/core/parcel/src/cli.js
index b65e2e1d2..bd9a7b3f4 100755
--- packages/core/parcel/src/cli.js
+++ packages/core/parcel/src/cli.js
@@ -93,12 +93,20 @@ const commonOptions = {
   ],
 };
 
+function parse(s) {
+  let v = Number(s);
+  if (!Number.isInteger(v)) {
+    throw new Error(`Defined port ${s} is not a valid integer`);
+  }
+  return v;
+}
+
 var hmrOptions = {
   '--no-hmr': 'disable hot module replacement',
   '-p, --port <port>': [
     'set the port to serve on. defaults to $PORT or 1234',
-    value => parseInt(value, 10),
-    process.env.PORT != null ? parseInt(process.env.PORT, 10) : undefined,
+    value => parse(value),
+    process.env.PORT != null ? parse(process.env.PORT) : undefined,
   ],
   '--host <host>':
     'set the host to listen on, defaults to listening on all interfaces',
@@ -345,11 +353,6 @@ async function normalizeOptions(command): Promise<InitialParcelOptions> {
   let serve = false;
   let {port = 1234, host} = command;
 
-  // Throw an error if port value is invalid...
-  if (!Number.isInteger(port)) {
-    throw new Error(`Defined port is not a valid integer`);
-  }
-
   // Ensure port is usable
   let originalPort = port;
   if (command.name() === 'serve' || command.hmr) {
```

Reply:
Sure, will refactor it

Reply:
Changed it, moved parsing into normaliseOptions because of not wanting uncaught exceptions